Data race in flaky CI test

### Checklist

- [x] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [x] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

The first run of the Ubuntu platform tests (platform_tests (ubuntu-latest)) for my PR #6345 failed with a data race.
This did not occur on re-run and I could not reproduce this in a local Ubuntu Docker container (`linux/amd64` on MacOS). So, the test is flaky but the data race output might be worth investigating.

Here’s the log (because no longer accessible via GHA after the re-run):

```
ok  	fyne.io/fyne/v2/internal/driver	0.152s	coverage: 90.7% of statements
ok  	fyne.io/fyne/v2/internal/driver/common	0.173s	coverage: 33.0% of statements
ok  	fyne.io/fyne/v2/internal/driver/embedded	0.280s	coverage: 15.6% of statements
==================
WARNING: DATA RACE
Read at 0x000002bf9a90 by main goroutine:
  fyne.io/fyne/v2/internal/async.(*Map[...]).Range()
      /home/runner/work/fyne/fyne/internal/async/map_migratedfynedo.go:54 +0x59
  fyne.io/fyne/v2/internal/cache.destroyExpiredSvgs()
      /home/runner/work/fyne/fyne/internal/cache/svg.go:47 +0x84
  fyne.io/fyne/v2/internal/cache.Clean()
      /home/runner/work/fyne/fyne/internal/cache/base.go:45 +0x184
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame()
      /home/runner/work/fyne/fyne/internal/driver/glfw/loop.go:98 +0xae
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL()
      /home/runner/work/fyne/fyne/internal/driver/glfw/loop.go:196 +0x49d
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run()
      /home/runner/work/fyne/fyne/internal/driver/glfw/driver.go:195 +0xce
  fyne.io/fyne/v2/internal/driver/glfw.TestMain()
      /home/runner/work/fyne/fyne/internal/driver/glfw/window_test.go:58 +0x13c
  main.main()
      _testmain.go:329 +0x404

Previous write at 0x000002bf9a90 by goroutine 128:
  fyne.io/fyne/v2/internal/async.(*Map[...]).Clear()
      /home/runner/work/fyne/fyne/internal/async/map_migratedfynedo.go:71 +0x3d
  fyne.io/fyne/v2/internal/cache.ResetThemeCaches()
      /home/runner/work/fyne/fyne/internal/cache/base.go:84 +0x56b
  fyne.io/fyne/v2/test.NewApp()
      /home/runner/work/fyne/fyne/test/app.go:203 +0x558
  fyne.io/fyne/v2/test.NewTempApp()
      /home/runner/work/fyne/fyne/test/app_helper.go:15 +0x2c
  fyne.io/fyne/v2/internal/driver/glfw.TestMenuBar_Toggle()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar_test.go:478 +0x4e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x47

Goroutine 128 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x75d
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1846 +0x99
  testing.tRunner()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1446 +0x216
  testing.runTests()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1844 +0x7ec
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1726 +0xa84
  fyne.io/fyne/v2/internal/driver/glfw.TestMain.func1()
      /home/runner/work/fyne/fyne/internal/driver/glfw/window_test.go:49 +0x4f
==================
==================
WARNING: DATA RACE
Write at 0x00c001219d40 by goroutine 129:
  runtime.mapassign()
      /opt/hostedtoolcache/go/1.19.13/x64/src/runtime/map.go:578 +0x0
  fyne.io/fyne/v2/internal/async.(*Map[...]).Store()
      /home/runner/work/fyne/fyne/internal/async/map_migratedfynedo.go:66 +0x176
  fyne.io/fyne/v2/internal/cache.SetFontMetrics()
      /home/runner/work/fyne/fyne/internal/cache/text.go:57 +0x213
  fyne.io/fyne/v2/internal/painter.RenderedTextSize()
      /home/runner/work/fyne/fyne/internal/painter/font.go:238 +0x265
  fyne.io/fyne/v2/test.(*driver).RenderedTextSize()
      /home/runner/work/fyne/fyne/test/driver.go:105 +0xb8
  fyne.io/fyne/v2/canvas.(*Text).MinSize()
      /home/runner/work/fyne/fyne/canvas/text.go:43 +0x12a
  fyne.io/fyne/v2/internal/driver/glfw.(*menuBarItemRenderer).MinSize()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar_item.go:152 +0x58
  fyne.io/fyne/v2/internal/widget.(*Base).MinSize()
      /home/runner/work/fyne/fyne/internal/widget/base.go:74 +0x5a
  fyne.io/fyne/v2/internal/driver/glfw.(*menuBarItem).MinSize()
      <autogenerated>:1 +0x37
  fyne.io/fyne/v2/layout.hBoxLayout.MinSize()
      /home/runner/work/fyne/fyne/layout/boxlayout.go:208 +0x13e
  fyne.io/fyne/v2.(*Container).MinSize()
      /home/runner/work/fyne/fyne/container.go:103 +0xf4
  fyne.io/fyne/v2/internal/driver/glfw.(*menuBarRenderer).MinSize()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar.go:154 +0x58
  fyne.io/fyne/v2/internal/widget.(*Base).MinSize()
      /home/runner/work/fyne/fyne/internal/widget/base.go:74 +0x5a
  fyne.io/fyne/v2/internal/driver/glfw.(*MenuBar).MinSize()
      <autogenerated>:1 +0x37
  fyne.io/fyne/v2.(*Container).MinSize()
      /home/runner/work/fyne/fyne/container.go:108 +0x196
  fyne.io/fyne/v2/test.(*canvas).SetContent()
      /home/runner/work/fyne/fyne/test/canvas.go:224 +0x224
  fyne.io/fyne/v2/test.(*window).SetContent()
      /home/runner/work/fyne/fyne/test/window.go:88 +0x5c
  fyne.io/fyne/v2/internal/driver/glfw.TestMenuBar_Toggle.func1()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar_test.go:497 +0x268
  testing.tRunner()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x47

Previous read at 0x00c001219d40 by main goroutine:
  runtime.mapiterinit()
      /opt/hostedtoolcache/go/1.19.13/x64/src/runtime/map.go:815 +0x0
  fyne.io/fyne/v2/internal/async.(*Map[...]).Range()
      /home/runner/work/fyne/fyne/internal/async/map_migratedfynedo.go:54 +0xa6
  fyne.io/fyne/v2/internal/cache.destroyExpiredFontMetrics()
      /home/runner/work/fyne/fyne/internal/cache/text.go:67 +0x84
  fyne.io/fyne/v2/internal/cache.Clean()
      /home/runner/work/fyne/fyne/internal/cache/base.go:46 +0x198
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame()
      /home/runner/work/fyne/fyne/internal/driver/glfw/loop.go:98 +0xae
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL()
      /home/runner/work/fyne/fyne/internal/driver/glfw/loop.go:196 +0x49d
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run()
      /home/runner/work/fyne/fyne/internal/driver/glfw/driver.go:195 +0xce
  fyne.io/fyne/v2/internal/driver/glfw.TestMain()
      /home/runner/work/fyne/fyne/internal/driver/glfw/window_test.go:58 +0x13c
  main.main()
      _testmain.go:329 +0x404

Goroutine 129 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x75d
  fyne.io/fyne/v2/internal/driver/glfw.TestMenuBar_Toggle()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar_test.go:490 +0x793
  testing.tRunner()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c001219d10 by goroutine 129:
  runtime.mapassign()
      /opt/hostedtoolcache/go/1.19.13/x64/src/runtime/map.go:578 +0x0
  fyne.io/fyne/v2/internal/async.(*Map[...]).Store()
      /home/runner/work/fyne/fyne/internal/async/map_migratedfynedo.go:66 +0x11e
  fyne.io/fyne/v2/internal/cache.SetSvg()
      /home/runner/work/fyne/fyne/internal/cache/svg.go:36 +0x195
  fyne.io/fyne/v2/canvas.(*Image).renderSVG()
      /home/runner/work/fyne/fyne/canvas/image.go:408 +0x250
  fyne.io/fyne/v2/canvas.(*Image).Refresh()
      /home/runner/work/fyne/fyne/canvas/image.go:172 +0x3ee
  fyne.io/fyne/v2/canvas.(*Image).Resize()
      /home/runner/work/fyne/fyne/canvas/image.go:219 +0x1eb
  fyne.io/fyne/v2/widget.(*menuItemRenderer).Layout()
      /home/runner/work/fyne/fyne/widget/menu_item.go:241 +0x375
  fyne.io/fyne/v2/internal/widget.(*Base).Resize()
      /home/runner/work/fyne/fyne/internal/widget/base.go:45 +0xea
  fyne.io/fyne/v2/widget.(*menuItem).Resize()
      <autogenerated>:1 +0x4f
  fyne.io/fyne/v2/layout.vBoxLayout.Layout()
      /home/runner/work/fyne/fyne/layout/boxlayout.go:106 +0x241
  fyne.io/fyne/v2.(*Container).layout()
      /home/runner/work/fyne/fyne/container.go:203 +0x159
  fyne.io/fyne/v2.(*Container).Resize()
      /home/runner/work/fyne/fyne/container.go:176 +0xaa
  fyne.io/fyne/v2/widget.(*menuBoxRenderer).Layout()
      /home/runner/work/fyne/fyne/widget/menu.go:347 +0x8f
  fyne.io/fyne/v2/widget.(*BaseWidget).Resize()
      /home/runner/work/fyne/fyne/widget/widget.go:51 +0xde
  fyne.io/fyne/v2/widget.(*menuBox).Resize()
      <autogenerated>:1 +0x4f
  fyne.io/fyne/v2/internal/widget.(*scrollContainerRenderer).Layout()
      /home/runner/work/fyne/fyne/internal/widget/scroller.go:418 +0xf4
  fyne.io/fyne/v2/internal/widget.(*Base).Resize()
      /home/runner/work/fyne/fyne/internal/widget/base.go:45 +0xea
  fyne.io/fyne/v2/internal/widget.(*Scroll).Resize()
      /home/runner/work/fyne/fyne/internal/widget/scroller.go:594 +0x93
  fyne.io/fyne/v2/widget.(*menuRenderer).Layout()
      /home/runner/work/fyne/fyne/widget/menu.go:251 +0x32b
  fyne.io/fyne/v2/widget.(*BaseWidget).Resize()
      /home/runner/work/fyne/fyne/widget/widget.go:51 +0xde
  fyne.io/fyne/v2/internal/driver/glfw.(*menuBarRenderer).Layout()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar.go:146 +0x3be
  fyne.io/fyne/v2/internal/driver/glfw.(*menuBarRenderer).Refresh()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar.go:158 +0x89
  fyne.io/fyne/v2/internal/widget.(*Base).Refresh()
      /home/runner/work/fyne/fyne/internal/widget/base.go:120 +0x58
  fyne.io/fyne/v2/internal/driver/glfw.(*MenuBar).activateChild()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar.go:90 +0x204
  fyne.io/fyne/v2/internal/driver/glfw.(*MenuBar).toggle()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar.go:115 +0xbe
  fyne.io/fyne/v2/internal/driver/glfw.(*MenuBar).Toggle()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar.go:65 +0x86
  fyne.io/fyne/v2/internal/driver/glfw.TestMenuBar_Toggle.func1()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar_test.go:504 +0x371
  testing.tRunner()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x47

Previous read at 0x00c001219d10 by main goroutine:
  runtime.mapiterinit()
      /opt/hostedtoolcache/go/1.19.13/x64/src/runtime/map.go:815 +0x0
  fyne.io/fyne/v2/internal/async.(*Map[...]).Range()
      /home/runner/work/fyne/fyne/internal/async/map_migratedfynedo.go:54 +0xa4
  fyne.io/fyne/v2/internal/cache.destroyExpiredSvgs()
      /home/runner/work/fyne/fyne/internal/cache/svg.go:47 +0x84
  fyne.io/fyne/v2/internal/cache.Clean()
      /home/runner/work/fyne/fyne/internal/cache/base.go:45 +0x184
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame()
      /home/runner/work/fyne/fyne/internal/driver/glfw/loop.go:98 +0xae
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL()
      /home/runner/work/fyne/fyne/internal/driver/glfw/loop.go:196 +0x49d
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run()
      /home/runner/work/fyne/fyne/internal/driver/glfw/driver.go:195 +0xce
  fyne.io/fyne/v2/internal/driver/glfw.TestMain()
      /home/runner/work/fyne/fyne/internal/driver/glfw/window_test.go:58 +0x13c
  main.main()
      _testmain.go:329 +0x404

Goroutine 129 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x75d
  fyne.io/fyne/v2/internal/driver/glfw.TestMenuBar_Toggle()
      /home/runner/work/fyne/fyne/internal/driver/glfw/menu_bar_test.go:490 +0x793
  testing.tRunner()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x000002bf9a70 by goroutine 128:
  fyne.io/fyne/v2/internal/async.(*Map[...]).Clear()
      /home/runner/work/fyne/fyne/internal/async/map_migratedfynedo.go:71 +0x3d
  fyne.io/fyne/v2/internal/cache.ResetThemeCaches()
      /home/runner/work/fyne/fyne/internal/cache/base.go:85 +0x57e
  fyne.io/fyne/v2/test.NewApp()
      /home/runner/work/fyne/fyne/test/app.go:203 +0x558
  fyne.io/fyne/v2/test.NewTempApp.func1()
      /home/runner/work/fyne/fyne/test/app_helper.go:16 +0x24
  testing.(*common).Cleanup.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1041 +0x193
  testing.(*common).runCleanup()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1210 +0x143
  testing.tRunner.func2()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1440 +0x52
  runtime.deferreturn()
      /opt/hostedtoolcache/go/1.19.13/x64/src/runtime/panic.go:476 +0x32
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x47

Previous read at 0x000002bf9a70 by main goroutine:
  fyne.io/fyne/v2/internal/async.(*Map[...]).Range()
      /home/runner/work/fyne/fyne/internal/async/map_migratedfynedo.go:54 +0x5c
  fyne.io/fyne/v2/internal/cache.destroyExpiredFontMetrics()
      /home/runner/work/fyne/fyne/internal/cache/text.go:67 +0x84
  fyne.io/fyne/v2/internal/cache.Clean()
      /home/runner/work/fyne/fyne/internal/cache/base.go:46 +0x198
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame()
      /home/runner/work/fyne/fyne/internal/driver/glfw/loop.go:98 +0xae
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL()
      /home/runner/work/fyne/fyne/internal/driver/glfw/loop.go:196 +0x49d
  fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run()
      /home/runner/work/fyne/fyne/internal/driver/glfw/driver.go:195 +0xce
  fyne.io/fyne/v2/internal/driver/glfw.TestMain()
      /home/runner/work/fyne/fyne/internal/driver/glfw/window_test.go:58 +0x13c
  main.main()
      _testmain.go:329 +0x404

Goroutine 128 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1493 +0x75d
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1846 +0x99
  testing.tRunner()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1446 +0x216
  testing.runTests()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1844 +0x7ec
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.19.13/x64/src/testing/testing.go:1726 +0xa84
  fyne.io/fyne/v2/internal/driver/glfw.TestMain.func1()
      /home/runner/work/fyne/fyne/internal/driver/glfw/window_test.go:49 +0x4f
==================
--- FAIL: TestMenuBar_Toggle (0.09s)
    --- FAIL: TestMenuBar_Toggle/when_menu_bar_is_inactive (0.08s)
        testing.go:1319: race detected during execution of test
    testing.go:1319: race detected during execution of test
FAIL
coverage: 69.3% of statements
FAIL	fyne.io/fyne/v2/internal/driver/glfw	22.693s
```

### How to reproduce

-

### Screenshots

_No response_

### Example code

-

### Fyne version

develop

### Go compiler version

1.19.13

### Operating system and version

Ubuntu 24.04.4

### Additional Information

_No response_

--- a/internal/driver/glfw/menu_bar_test.go
+++ b/internal/driver/glfw/menu_bar_test.go
@@ -17,7 +17,9 @@ import (
 )
 
 func TestMenuBar(t *testing.T) {
-	test.NewTempApp(t)
+	runOnMain(func() {
+		test.NewTempApp(t)
+	})
 
 	var lastAction string
 
@@ -475,7 +477,9 @@ func TestMenuBar(t *testing.T) {
 }
 
 func TestMenuBar_Toggle(t *testing.T) {
-	test.NewTempApp(t)
+	runOnMain(func() {
+		test.NewTempApp(t)
+	})
 
 	m1i1 := fyne.NewMenuItem("New", nil)
 	m1i2 := fyne.NewMenuItem("Open", nil)

